app/main.py

```diff
@@ -2,5 +2,14 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_store = {}
+
+    def inner(*args) -> Callable:
+        if args in cache_store:
```

Comment:
Using `args` as a key in `cache_store` can lead to issues if any of the arguments are mutable (e.g., lists or dictionaries). Mutable types are not hashable and will raise a `TypeError` if used as a dictionary key. Consider adding a check to ensure all arguments are hashable or converting them to a hashable type (e.g., using a tuple of tuples for dictionaries).